map/gps_track.cpp

```diff
@@ -351,3 +359,27 @@ void GpsTrack::NotifyCallback(pair<size_t, size_t> const & addedIds, pair<size_t
     m_callback(std::move(toAdd), evictedIds);
   }
 }
+
+std::vector<location::GpsTrackInfo> GpsTrack::GetCurrentTrack() const
+{
+  lock_guard<mutex> lg(m_dataGuard);
+  if (!m_collection)
+    return {};
+
+  std::vector<location::GpsTrackInfo> track;
+  track.reserve(m_collection->GetSize());
+  m_collection->ForEach([&track](location::GpsTrackInfo const & point, size_t id)->bool
+  {
+    track.emplace_back(point);
+    return true;
+  });
+
+  return track;
+}
+
+bool GpsTrack::IsCollectionInit() const
+{
+  lock_guard<mutex> lg(m_dataGuard);
```

Comment:
1. Is data guard really needed here?
2. Why is this method needed at all?

```suggestion
  std::lock_guard lg(m_dataGuard);
```